src/ui/components/ExplorerHome/ExplorerHome.js

```diff
@@ -244,40 +238,37 @@ export const ExplorerHome = ({
 
   // build structures for timelines at the top of the page
   const {credTotalsTimeline, grainTotalsTimeline} = useMemo(() => {
-    const allParticipantsCred: Array<Array<number>> = allParticipants.map(
-      (participant) => {
-        return participant.credPerInterval.slice();
-      }
-    );
-    const allParticipantsGrain: Array<Array<Grain>> = allParticipants.map(
-      (participant) => {
-        return participant.grainEarnedPerInterval.slice();
-      }
-    );
-    const credAccumulator = (
-      credTimelineAccumalator: Array<number>,
-      participantCred
-    ) =>
-      credTimelineAccumalator.map(
-        (intervalTotal, i) => intervalTotal + participantCred[i]
-      );
-    const grainAccumulator = (
-      grainTimelineAccumulator: Array<Grain>,
-      participantGrain
-    ) =>
-      grainTimelineAccumulator.map((intervalTotal, i) =>
-        add(intervalTotal, participantGrain[i])
-      );
-    const credTimeline: Array<number> = allParticipantsCred.reduce(
-      credAccumulator
-    );
-    const grainTimeline: Array<Grain> = allParticipantsGrain.reduce(
-      grainAccumulator
-    );
-    return {
-      credTotalsTimeline: credTimeline,
-      grainTotalsTimeline: grainTimeline,
+    const totals = {
+      credTotalsTimeline: [],
+      grainTotalsTimeline: [],
     };
+
+    return allParticipants.reduce(
+      ({credTotalsTimeline, grainTotalsTimeline}, participant) => {
+        const {credPerInterval, grainEarnedPerInterval} = participant;
+        // find the longer array (if different) and iterate over both simultaneously
+        const length = Math.max(
+          credPerInterval.length,
+          grainEarnedPerInterval.length
+        );
+
+        for (let i = 0; i < length; i++) {
+          credTotalsTimeline[i] =
+            credPerInterval[i] + credTotalsTimeline[i] || 0;
+
+          grainTotalsTimeline[i] = add(
+            grainEarnedPerInterval[i],
+            grainTotalsTimeline[i]
+          );
+        }
```

Comment:
This confuses me. I believe it's a true assumption that the length of the arrays should always be equal and the elements should always be present as a number/Grain value

Reply:
> This confuses me. I believe it's a true assumption that the length of the arrays should always be equal and the elements should always be present as a number/Grain value

@blueridger I also believe the arrays being the same length and the values present is a valid assumption. My intention in including the `Math.max()` is defensive programming. If there were to be a case in which, say, one array were populated and the other zero-length, it wouldn't cause issues of valid data going missing. Do you think this scenario is too far-fetched to bother putting in a guard for? Or perhaps a better comment would make it less confusing?

Reply:
I think we can trust that assumption and remove the guard rail. Also, why doesn't 261 have an or zero like 257? I'm guessing Flow is not considering the index out of bounds possibility and bigInt is simply casting undefined as 0. Just surprising and may be worth making the zero conversion explicit.

Reply:
Ok sounds good. 261 doesn't have an `|| 0` because, as you rightly surmised, it casts undefined as 0. However, this is only the behavior of the bigInt library, and not `BigInt` native implementation, which throws on `BigInt(undefined)`. I discovered this today in digging into PR # 2821. For forward compatibility, an `|| 0` guard is a good idea and I'll add it. I think flow didn't complain because the bigInt library isn't typed.